src/averageAges.js

```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(({ sex, name }) =>
+    withChildren
+      ? sex === 'f' && people.some(({ mother }) => mother === name)
+      : sex === 'f'
+  );
```

Comment:
I see that you repeat `sex` check second time already, can we make some type of function that'll filter people by sex for us?